### PR TITLE
SWC-7000: fix logical error

### PR DESCRIPTION
**Pull request by jay-hodgson**

### CODE REVIEW

File: src/main/java/org/sagebionetworks/web/client/presenter/TeamPresenter.java [hunk 1]
```diff
@@ -61,7 +61,6 @@ public class TeamPresenter
   private IsACTMemberAsyncHandler isACTMemberAsyncHandler;
   private TeamProjectsModalWidget teamProjectsModalWidget;
   private PortalGinInjector ginInjector;
-  private FeatureFlagConfig featureFlagConfig;
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/presenter/TeamPresenter.java:64]: Clean up member variable, which is not used (config is only referenced during construction)

File: src/main/java/org/sagebionetworks/web/client/presenter/TeamPresenter.java [hunk 2]
```diff
@@ -229,6 +227,7 @@ public void onSuccess(TeamBundle result) {
           view.setTeam(team, teamMembershipStatus);
           Boolean canRequestMembership = team.getCanRequestMembership();
           boolean isLockedDown =
+            !canPublicJoin &&
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/presenter/TeamPresenter.java:230]: Team is not logged down if public can join!

File: src/main/java/org/sagebionetworks/web/client/widget/team/controller/TeamEditModalWidget.java [hunk 3]
```diff
@@ -159,18 +159,23 @@ private void show() {
     view.clear();
     view.configure(team);
     //SWC-6259
-    Boolean canPublicJoin = team.getCanPublicJoin();
+    //SWC-7000
+    Boolean canPublicJoin = team.getCanPublicJoin() == null
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/widget/team/controller/TeamEditModalWidget.java:163]: Match logic to TeamPresenter.  canPublicJoin is true iff team.getCanPublicJoin().
Is locked down only if !canPublicJoin and team.getCanRequestMembership()===false

File: src/test/java/org/sagebionetworks/web/unitclient/widget/entity/team/controller/TeamEditModalWidgetTest.java [hunk 4]
```diff
@@ -431,6 +431,16 @@ public void testConfigurePublicCanJoin() {
     verify(mockTeam).setCanRequestMembership(false);
   }
 
+  @Test
+  public void testConfigurePublicCanJoinCanRequestMembershipFalse() {
```
Comment (jay-hodgson) [inline src/test/java/org/sagebionetworks/web/unitclient/widget/entity/team/controller/TeamEditModalWidgetTest.java:435]: All existing tests still pass.  Add test for failure case, when canRequestMembership == false (not null).  This failed before the change, and passes now.

File: src/main/java/org/sagebionetworks/web/client/widget/team/controller/TeamEditModalWidget.java (hunk 3 above)
Comment (nickgros) [inline src/main/java/org/sagebionetworks/web/client/widget/team/controller/TeamEditModalWidget.java:163]: These should never be null since https://sagebionetworks.jira.com/browse/PLFM-8304 was done, are you seeing a regression?

**Review by nickgros — APPROVED** [review]
